modules/dhcp/providers/server/infoblox.rb

```diff
@@ -0,0 +1,148 @@
+      logger.debug "Add Record - options[:nextServer] #{options[:nextServer]}, options[:filename] #{options[:filename]}, hostip.ipv4addr: #{hostip.ipv4addr} "
+      raise InvalidRecord, "#{record} Hostname mismatch" unless hostip.host == record.name
+      hostip.mac = record.mac
+      hostip.configure_for_dhcp = true
+      hostip.nextserver = options[:nextServer]
+      hostip.use_nextserver = true
+      hostip.bootfile = options[:filename]
+      hostip.use_bootfile = true
+      ## Test if Host Entry has correct IP
+      raise InvalidRecord, "#{record} IP mismatch" unless hostip.ipv4addr == record.ip
+      # Send object
+      post ? host.post : host.put
+      record
+    end
+    
+    def find_record(record)
+      logger.debug "loadRecord"
+      connection = connect
+      # if record is a String it can be either ip or mac, true = mac --> lookup ip
+      if record.is_a?(String) && (IPAddr.new(record) rescue nil).nil?
+        hostdhcp = ::Infoblox::HostIpv4addr.find(connection, "mac" => record).first
+        ipv4address = hostdhcp.ipv4addr
+      elsif record.is_a?(String)
+        ipv4address = record
+      end
+      ipv4address = record[:ip] if record.is_a?(Proxy::DHCP::Record)
+      ipv4address = record.to_s if record.is_a?(IPAddr)
+      host = ::Infoblox::Host.find(connection, "ipv4addr" => ipv4address).first
+      return nil if host.nil? || host.name.empty?
+      hostdhcp = ::Infoblox::HostIpv4addr.find(connection, "ipv4addr" => ipv4address).first
+      return nil unless hostdhcp.configure_for_dhcp
+      return nil if hostdhcp.mac.empty? || hostdhcp.ipv4addr.empty?
+      opts = {:hostname => host.name}
+      opts[:mac] = hostdhcp.mac
+      opts[:ip] = hostdhcp.ipv4addr
+      opts[:deleteable] = true
+      opts[:nextServer] = hostdhcp.nextserver if hostdhcp.use_nextserver
+      opts[:filename] = hostdhcp.bootfile if hostdhcp.use_bootfile
+      # Subnet should only be one, not checking that yet
+      subnet = subnets.find { |s| s.include? ipv4address}
+      Proxy::DHCP::Record.new(opts.merge(:subnet => subnet))
+    end
+
+    def loadSubnetData subnet
+      # Load network from infoblox, iterate over ips to gather additional settings
+      logger.debug "LoadSubnetData"
+      super
+      network = IPAddr.new(subnet.to_s, Socket::AF_INET)
+      connection = connect
+      # max results are currently set to work in my setup, one could calculate that setting by looking at netmask :)
+      network = ::Infoblox::Ipv4address.find(connection, "network" => "#{network}/#{subnet.cidr}", "_max_results" => "2500")
+      # Find out which hosts are in use
+      network.each do |host|
+        # next if certain values are not set
+        next if host.names.empty? || host.mac_address.empty? || host.ip_address.empty?
+        hostdhcp = ::Infoblox::HostIpv4addr.find(connection, "ipv4addr" => host.ip_address).first
```

Comment:
I did a quick look at infoblox api, but couldn't find any bulk operations, is that true? You may want to check if hostdhcp isn't empty -- if the number of records is large and leases short, there's a chance that the record may not be there by the time you get here.